api/server/errorhandler.go

```diff
@@ -17,18 +22,38 @@ func makeErrorHandler(err error) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		statusCode := httpstatus.FromError(err)
 		vars := mux.Vars(r)
-		if apiVersionSupportsJSONErrors(vars["version"]) {
-			response := &types.ErrorResponse{
-				Message: err.Error(),
-			}
+		version := vars["version"]
+
+		errMsg := err.Error()
+		stringErrs := unwrapErrors(err, true)
+		if len(stringErrs) > 1 {
+			errMsg = fmt.Sprintf("%d errors occurred:\n\t* %s", len(stringErrs), strings.Join(stringErrs, "\n\t* "))
+		}
+
+		const firstAPIVersionWithJSONErrors = "1.24"
+		const firstAPIVersionWithJSONMultiErrors = "1.44"
+
+		switch {
+		case version == "" || versions.GreaterThanOrEqualTo(version, firstAPIVersionWithJSONMultiErrors):
+			response := &types.ErrorResponse{Message: errMsg, RawMessages: stringErrs}
 			_ = httputils.WriteJSON(w, statusCode, response)
-		} else {
+		case versions.GreaterThanOrEqualTo(version, firstAPIVersionWithJSONErrors):
+			response := &v1p43.ErrorResponse{Message: errMsg}
+			_ = httputils.WriteJSON(w, statusCode, response)
+		default:
 			http.Error(w, status.Convert(err).Message(), statusCode)
 		}
 	}
 }
 
-func apiVersionSupportsJSONErrors(version string) bool {
-	const firstAPIVersionWithJSONErrors = "1.23"
-	return version == "" || versions.GreaterThan(version, firstAPIVersionWithJSONErrors)
+func unwrapErrors(err error, unwrapHTTP bool) []string {
+	var stringErrs []string
+	if errs, ok := err.(interface{ Unwrap() []error }); ok {
+		for _, err := range errs.Unwrap() {
+			stringErrs = append(stringErrs, err.Error())
+		}
+	} else if unwrapHTTP && errdefs.IsHTTPError(err) {
+		return unwrapErrors(errors.Unwrap(err), false)
```

Comment:
I think unwrapping only once is enough but I might be wrong. /cc @thaJeztah 

Also, this makes me wonder whether the fallback format should be handled by our own implementation of `joinError`. Otherwise, if we start using an Error tree, only the root will be well-formatted and other levels will use the line break format, as can be seen here: https://go.dev/play/p/MCUaNfjd5Xx.